crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -189,6 +275,46 @@ impl TransactionsForAccount for PendingTransactionsForAccount {
         // is equal to the account nonce
         self.txs().contains_key(&previous_nonce) || ttx.signed_tx.nonce() == current_account_nonce
     }
+
+    fn has_balance_to_cover(
+        &self,
+        ttx: &TimemarkedTransaction,
+        current_account_balances: &HashMap<IbcPrefixed, u128>,
+    ) -> bool {
+        // build up transaction cost map
+        let mut transaction_costs = HashMap::<IbcPrefixed, u128>::new();
```

Comment:
I included everything but the `CostDeductionError` part of the suggestion. It would be worthwhile to give users a comprehensive way to see their balances and pending/parked statuses outside of this PR. 

I think creating this error category here would be too heavy weight since the errors as is would only return partial information on how much funds are lacking

Reply:
Ah, right.  I think I was originally expecting we could use this method in what is now named `subtract_contained_costs` and there we would have wanted to match on the error variants to log the appropriate thing, so `anyhow::Error` wouldn't have been good for that.

Seems like we actually don't use the returned error anywhere other than for checking `.is_err()` or `.is_ok()`, so the actual info added in the `anyhow!` calls won't be visible to anyone as things stand, but that's not a problem I guess :)